src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.cs

```diff
@@ -545,244 +545,1012 @@ public abstract class Avx2 : Avx
         /// <summary>
         /// __m128i _mm_i32gather_epi32 (int const* base_addr, __m128i vindex, const int scale)
         ///   VPGATHERDD xmm, vm32x, xmm
+        /// The scale parameter should be 1, 2, 4 or 8, otherwise, ArgumentOutOfRangeException will be thrown.
         /// </summary>
-        public static unsafe Vector128<int> GatherVector128(int* baseAddress, Vector128<int> index, byte scale) => GatherVector128(baseAddress, index, scale);
+        public static unsafe Vector128<int> GatherVector128(int* baseAddress, Vector128<int> index, byte scale)
+        {
+            switch (scale)
+            {
+                case 1:
+                    return GatherVector128(baseAddress, index, 1);
+                case 2:
+                    return GatherVector128(baseAddress, index, 2);
+                case 4:
+                    return GatherVector128(baseAddress, index, 4);
+                case 8:
+                    return GatherVector128(baseAddress, index, 8);
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
nit: Should be `throw new ArgumentOutOfRangeException(nameof(scale))`

Reply:
(same elsewhere as well)

Reply:
Other intrinsic throw the exception without the argument name. Shall we keep consistent?

Reply:
Agree, logged at https://github.com/dotnet/coreclr/issues/19689